core/src/calleeFunctions/UniswapV3Callee.ts

```diff
@@ -39,20 +37,18 @@ const getMarketPrice = async function (
     network: string,
     collateral: CollateralConfig,
     marketId: string,
-    collateralAmount: BigNumber
+    collateralAmount: BigNumber,
+    preloadedPools?: Pool[]
 ): Promise<BigNumber> {
-    // convert collateral into DAI
-    const { route, fees } = await getRouteAndGasQuote(network, collateral.symbol, collateralAmount, marketId);
-    if (!route) {
-        throw new Error(`No route found for ${collateral.symbol} to DAI`);
+    if (!preloadedPools) {
+        throw new Error(`pools required to get market price for callee type "${marketId}"`);
```

Comment:
this intends to address https://github.com/sidestream-tech/unified-auctions-ui/pull/531#discussion_r1039906540